src/go/configgenerator/route_generator.go

```diff
@@ -239,11 +238,16 @@ func makeHttpRouteMatcher(httpRule *commonpb.Pattern) *routepb.RouteMatch {
 		return nil
 	}
 	var routeMatcher routepb.RouteMatch
-	re := regexp.MustCompile(`{[^{}]+}`)
 
-	// Replace path templates inside "{}" by regex "[^\/]+", which means
-	// any character except `/`, also adds `$` to match to the end of the string.
-	if re.MatchString(httpRule.UriTemplate) {
+	regex, err := util.GetRegexForUriWithPathParams(httpRule.UriTemplate)
```

Comment:
So wildcard is the only case of using regex here?

Reply:
Can we comment all the possible route cases here?

Reply:
- Changed function to return single string, no error or boolean needed.
- Renamed it, in case it was not clear.
- No, this method supports any URI that has path parameters. See the test. It has nothing to do with wildcards inside the path params, instead the path parameters should be treated as wildcards.